### PR TITLE
Expand ModeMonitor docstring and add changelog explanation for backward mode sign change

### PR DESCRIPTION
**Pull request by momchil-flex**

### CODE REVIEW

**Comment by momchil-flex** [comment]
Not sure if I phrased it right, let me know what you think.

**Review by tylerflex — APPROVED** [review]
I think these are all pretty clear, thanks

**Comment by momchil-flex** [comment]
Also updated the mode monitor notebook: https://github.com/flexcompute-readthedocs/tidy3d-docs/commit/8cb6d4978e04dfa0a4221ab7e33177fb22465237

File: tidy3d/components/monitor.py [hunk 1]
```diff
@@ -456,6 +456,11 @@ def storage_size(self, num_cells: int, tmesh: ArrayFloat1D) -> int:
 
 class ModeMonitor(AbstractModeMonitor):
     """:class:`Monitor` that records amplitudes from modal decomposition of fields on plane.
+    The amplitudes are defined as
+    ``recorded_field.dot(mode_solver_data) / mode_solver_data.dot(mode_solver_data)``, where
```
Comment (dbochkov-flexcompute) [inline tidy3d/components/monitor.py:460]: it's actually `mode_solver_data.dot(recorded_field) / mode_solver_data.dot(mode_solver_data)` but I don't think it matters

**Review by dbochkov-flexcompute — APPROVED** [review]

File: tidy3d/components/monitor.py (hunk 1 above)
Comment (momchil-flex) [reply tidy3d/components/monitor.py:460]: Thanks, I'll change anyway.

**Pull request merged**